<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1187</width>
    <height>670</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modbus</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_8">
    <item row="0" column="0">
     <widget class="QGroupBox" name="ConnectionBox">
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>150</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>280</width>
        <height>150</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Icons</family>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="title">
       <string>Connection</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_10">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="ConnectionStatus">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>18</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>DISCONNECTED</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLineEdit" name="ipText">
         <property name="inputMethodHints">
          <set>Qt::ImhDigitsOnly|Qt::ImhUrlCharactersOnly</set>
         </property>
         <property name="text">
          <string>172.23.0.8</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="PortText">
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>502</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QPushButton" name="ConnectButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>12</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" rowspan="2">
     <widget class="QGroupBox" name="Data">
      <property name="minimumSize">
       <size>
        <width>600</width>
        <height>300</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Icons</family>
       </font>
      </property>
      <property name="title">
       <string>Data</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_13">
       <item row="0" column="4">
        <widget class="QLabel" name="HotelLabel">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>Hotel</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="DeviceLabel">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>Device Code</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="QLabel" name="OperationLabel">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>Operation</string>
         </property>
        </widget>
       </item>
       <item row="6" column="5">
        <widget class="QSpinBox" name="OpData5">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item row="1" column="5">
        <widget class="QSpinBox" name="OpData3">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item row="7" column="4">
        <widget class="QPushButton" name="QueueButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Queue</string>
         </property>
        </widget>
       </item>
       <item row="3" column="5" rowspan="2">
        <widget class="QSpinBox" name="OpData4">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0" rowspan="6" colspan="2">
        <widget class="QListWidget" name="DeviceCode">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>Invalid Position</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Hotel</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Storage</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>ManLoadIN</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>LiquidHandling</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>FluorCam</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>HC1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>HC2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>ManLoadOUT</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Trash</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>SafetyPlace</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="4" rowspan="2">
        <widget class="QLabel" name="DrawerLabel">
         <property name="text">
          <string>Drawer</string>
         </property>
        </widget>
       </item>
       <item row="4" column="4" rowspan="3">
        <widget class="QListWidget" name="OpData2">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="7" column="1" colspan="2">
        <widget class="QPushButton" name="SetButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Set Data</string>
         </property>
        </widget>
       </item>
       <item row="2" column="5">
        <widget class="QLabel" name="GrabLabel">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>Grab Height</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2" rowspan="6" colspan="2">
        <widget class="QListWidget" name="OpCode">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>Invalid operation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>PickUp</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Place</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Tag read</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>IsEmpty</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Calibration</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="5" column="5">
        <widget class="QLabel" name="AddLabel">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>Additional</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QListWidget" name="OpData1">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="7" column="3">
        <widget class="QPushButton" name="NextButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Next</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QPushButton" name="ExecuteButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Execute</string>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QLabel" name="PosLabel">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>Position</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="2" rowspan="3">
     <widget class="QGroupBox" name="ControlBox">
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Icons</family>
       </font>
      </property>
      <property name="title">
       <string>Controls</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_14">
       <item row="4" column="3" colspan="2">
        <widget class="QSpinBox" name="spinBox"/>
       </item>
       <item row="0" column="0" colspan="2">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="RunLabel">
           <property name="font">
            <font>
             <family>Icons</family>
             <pointsize>9</pointsize>
             <underline>false</underline>
            </font>
           </property>
           <property name="text">
            <string>Not Running</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="ErrLabel">
           <property name="text">
            <string>Error:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="PlusButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>+</string>
         </property>
         <property name="autoDefault">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2">
        <widget class="QPushButton" name="MinusButton">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>-</string>
         </property>
         <property name="autoDefault">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="LoadQueueButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Load</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="StartQueueButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1" colspan="2">
        <widget class="QPushButton" name="StopQueueButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="5">
        <widget class="QListWidget" name="QueueList"/>
       </item>
       <item row="3" column="3" colspan="2">
        <widget class="QCheckBox" name="checkBox">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="text">
          <string>CheckBox</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QPushButton" name="StopButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="2">
        <widget class="QLabel" name="SpeedLabel">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>15</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Speed:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QLCDNumber" name="SpeedDisplay">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>72</pointsize>
          </font>
         </property>
         <property name="digitCount">
          <number>3</number>
         </property>
         <property name="segmentStyle">
          <enum>QLCDNumber::Flat</enum>
         </property>
         <property name="value" stdset="0">
          <double>0.000000000000000</double>
         </property>
         <property name="intValue" stdset="0">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item row="4" column="1" colspan="2">
        <widget class="QPushButton" name="SaveQueueButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="QPushButton" name="PlayButton">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>11</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Start
Pause</string>
         </property>
         <property name="autoDefault">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0" rowspan="2">
     <widget class="QGroupBox" name="RobotStatesBox">
      <property name="maximumSize">
       <size>
        <width>280</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Robot States</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_11">
       <item row="0" column="0">
        <widget class="QListWidget" name="StateList">
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QGroupBox" name="DataReadBox">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>250</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>350</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Icons</family>
       </font>
      </property>
      <property name="title">
       <string>Registers</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_12">
       <item row="0" column="0">
        <widget class="QListWidget" name="RegistersName">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QListWidget" name="RegistersData">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QListWidget" name="RegistersDataNames">
         <property name="font">
          <font>
           <family>Icons</family>
          </font>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QListWidget" name="RegistersReadName">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>10</pointsize>
          </font>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QListWidget" name="RegistersReadData">
         <property name="font">
          <font>
           <family>Icons</family>
           <pointsize>10</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
